### DISCUSSION TITLE
Use a different json parser to allow trailing commas

### DISCUSSION THREAD

**Discussion by ianldgs** [nrwl/nx#6711] [Feature Requests]
Instead of parsing the nx.json, workspace.json and tsconfigs with a regular JSON parser, NX could use the same parser that TypeScript uses by default, allowing trailing commas and preventing conflicts on those files if multiple pull requests add a new library at the same time. Or at least the conflict will be easier to resolve. It would also allow comments on the JSON file, which is something that the regular JSON parser doesn't allow.
Any thoughts? :)

**Comment by tyler-lemke** [comment]
Trailing commas caused errors for us during our current migration from Nx 13 to Nx 14.2. It would be nice to prevent this happening on our next migration.
